autorest/azure/cli/token.go

```diff
@@ -173,3 +181,43 @@ func GetTokenFromCLI(resource string) (*Token, error) {
 
 	return &tokenResponse, err
 }
+
+func validateParameter(param string) error {
+	// Validate parameters, since it gets sent as a command line argument to Azure CLI
+	const invalidResourceErrorTemplate = "Parameter %s is not in expected format. Only alphanumeric characters, [dot], [colon], [hyphen], and [forward slash] are allowed."
+	match, err := regexp.MatchString("^[0-9a-zA-Z-.:/]+$", param)
+	if err != nil {
+		return err
+	}
+	if !match {
+		return fmt.Errorf(invalidResourceErrorTemplate, param)
+	}
+	return nil
+}
+
+// GetAzureCLICommand can be used to run arbitrary Azure CLI command
+func GetAzureCLICommand() *exec.Cmd {
```

Comment:
Does this need to be exported?

Reply:
Yes. I want to run `az login` commands when it can't get credentials.